cpp/include/cuspatial/detail/index/construction/phase_2.cuh

```diff
@@ -85,9 +85,9 @@ inline rmm::device_uvector<uint32_t> flatten_point_keys(
                     keys_and_levels + num_valid_nodes,
                     flattened_keys.begin(),
                     [last_level = max_depth - 1] __device__(auto const& val) {
-                      bool is_parent{false};
-                      uint32_t key{}, level{};
-                      thrust::tie(key, level, is_parent) = val;
+                      auto& key       = thrust::get<0>(val);
+                      auto& level     = thrust::get<1>(val);
+                      auto& is_parent = thrust::get<2>(val);
```

Comment:
structured binding didn't work?

Reply:
Structured binding works with new CCCL 2.5, but not current CCCL 2.2.